frontend/src/GameReportForm.tsx

```diff
@@ -218,224 +397,317 @@ function GameReportForm() {
+                    error={formData.aragornCrownedTurn.error}
+                >
                     <SelectNumericOptionInput
                         start={1}
                         end={18}
                         onChange={handleInputChange("aragornCrownedTurn")}
+                        validate={validateField("aragornCrownedTurn")}
                     />
                 </GameReportFormElement>
             )}
             <GameReportFormElement
                 label={"What strongholds were captured by Shadow?"}
+                error={formData.capturedStrongholds.error}
             >
                 <VictoryPoints
-                    strongholds={formData.capturedStrongholds}
+                    strongholds={formData.capturedStrongholds.value}
                 ></VictoryPoints>
                 <MultiOptionInput
                     values={strongholds.slice()}
-                    current={formData.capturedStrongholds}
+                    current={formData.capturedStrongholds.value}
                     onChange={handleInputChange("capturedStrongholds")}
+                    validate={validateField("capturedStrongholds")}
                 />
             </GameReportFormElement>
             <GameReportFormElement
                 label={
                     "On a scale from 1-10, how interesting did you find this game?"
                 }
+                error={formData.interestRating.error}
             >
                 <SelectNumericOptionInput
                     start={1}
                     end={10}
                     onChange={handleInputChange("interestRating")}
+                    validate={validateField("interestRating")}
                 />
             </GameReportFormElement>
             <GameReportFormElement
                 label={"Do you have any comments or questions?"}
+                error={formData.comment.error}
             >
                 <TextInput
-                    value={formData.comment || ""}
+                    value={formData.comment.value || ""}
                     placeholder=""
                     onChange={handleInputChange("comment")}
+                    validate={validateField("comment")}
                 />
             </GameReportFormElement>
             <Button onClick={handleSubmit}>Submit</Button>
+            {errorOnSubmit && (
+                <Typography color="danger">{errorOnSubmit}</Typography>
+            )}
         </Sheet>
     );
 }
 
 interface GameReportElementProps {
     children: React.ReactNode;
     label: string;
+    error?: FieldError;
 }
 
-function GameReportFormElement({ children, label }: GameReportElementProps) {
+function GameReportFormElement({
+    children,
+    label,
+    error,
+}: GameReportElementProps) {
     return (
         <Sheet variant="outlined" sx={{ p: 2, borderRadius: "lg" }}>
-            <FormLabel sx={{ fontSize: 16, pb: 2 }}>{label}</FormLabel>
-            {children}
+            <FormControl error={!!error}>
```

Comment:
I added a `FormControl` wrapper to take advantage of MUI's default error styling, but it does introduce a console error `A FormControl can contain only one control component...` because of how one of the multiselect list components is implemented. Want to fix separately so I don't mix in a lot of UI changes